contracts/container/contract.go

```diff
@@ -238,6 +239,42 @@ func Update(script []byte, manifest []byte, data any) {
 	runtime.Log("container contract updated")
 }
 
+// PutObject registers successful object PUT operation and notifies about
+// it. metaInformation must be signed by container nodes according to
+// container's placement, see [VerifyPlacementSignatures]. metaInformation
+// must contain information about an object placed to a container that was
+// created with meta-on-chain feature enabled.
+func PutObject(metaInformation []byte, sigs [][]interop.Signature) {
+	metaMap := std.Deserialize(metaInformation).(map[string]any)
+	cID := metaMap["cid"].(interop.Hash256)
+	if len(cID) != interop.Hash256Len {
+		panic("incorrect container ID")
+	}
+
+	if storage.Get(storage.GetContext(), append([]byte{containersWithMetaPrefix}, cID...)) == nil {
+		panic("container does not support meta-on-chain")
+	}
+
+	if !VerifyPlacementSignatures(cID, metaInformation, sigs) {
+		panic("signature verification failed")
+	}
+
+	oid := metaMap["oid"].(interop.Hash256)
+	runtime.Notify("ObjectPut", cID, oid)
```

Comment:
@roman-khimov, see if it suits your expectations about notifications. on one hand, it is ok to know only cid/oid, on the other hand, why we have this info if it is not pushed to the notification? but also, we have a limitation on notification length, so no 1000 object deletion at a time and notification about it. also, we may have more fields in the future, should we change notification interface then?

Reply:
You have this call per event (object) anyway. And we need max data possible, so maybe pushing the whole map is appropriate here. 1K size is per even also, so it won't be a problem.

Reply:
> 1K size is per even also, so it won't be a problem.

but we can delete 1k objects with a single tombstone object, this is 1k hashes according to https://github.com/nspcc-dev/neofs-node/blob/7b592bdb1636bb67f55100877b3bf72fcc43ff4a/pkg/core/object/replicate.go#L35

Reply:
We need more strict limits there then. We no longer _need_ this for split objects, so I'd even think of 1:1 mapping (single tombstone --- single object).

Reply:
https://github.com/nspcc-dev/neofs-api/issues/311